common/predictive-text/unit_tests/in_browser/cases/worker-dummy-integration.js

```diff
@@ -0,0 +1,154 @@
+var assert = chai.assert;
+
+/*
+ * Shows off the LMLayer API, using the full prediction interface.
+ * The dummy model (or mock model) is a language model which has 
+ * **injectable** suggestions: that is, you, as the tester, have
+ * to provide the predictions. The dummy model does not create any
+ * suggestions on its own. The dummy model can take in a series
+ * of suggestions when initialized and return them sequentially.
+ */
+describe('LMLayer using dummy model', function () {
+  describe('Prediction', function () {
+    it('will predict future suggestions', function () {
+      var lmLayer = new LMLayer();
+      var capabilities = {
+        maxLefContextCodeUnits: 32 + ~~Math.random() * 32
+      };
+
+      // Tower of promises!
+      return lmLayer.initialize(
+        capabilities,
+        {
+            type: 'dummy',
+            futureSuggestions: iGotDistractedByHazel()
+        }
+      ).then(function (actualConfiguration) {
+        return Promise.resolve();
+      }).then(function () {
+        return lmLayer.predict(zeroTransform(), emptyContext());
+      }).then(function (suggestions) {
+        assert.deepEqual(suggestions, iGotDistractedByHazel()[0]);
+        return lmLayer.predict(zeroTransform(), emptyContext());
+      }).then(function (suggestions) {
+        assert.deepEqual(suggestions, iGotDistractedByHazel()[1]);
+        return lmLayer.predict(zeroTransform(), emptyContext());
+      }).then(function (suggestions) {
+        assert.deepEqual(suggestions, iGotDistractedByHazel()[2]);
+        return lmLayer.predict(zeroTransform(), emptyContext());
+      }).then(function (suggestions) {
+        assert.deepEqual(suggestions, iGotDistractedByHazel()[3]);
+        return Promise.resolve();
+      });
+    });
+  });
+
+  function emptyContext() {
+    return { left: '', startOfBuffer: true, endOfBuffer: true };
+  }
+
+  function zeroTransform() {
+    return { insert: '', deleteLeft: 0 };
+  }
+
+  function iGotDistractedByHazel() {
+    return [
```

Comment:
I see that we're defining this twice, once per utilizing test case.  You may wish to examine the idea of JSON fixtures - just save the JS objects as JSON, then use Karma to auto-load 'em and link them in.  That'll make it easier to share and maintain potential tests.

Reply:
Took some doing, but I got it! I'm thinking of also including this in the headless tests run by Mocha on Node. Does that sound reasonable?